src/main/java/com/razorpay/ApiClient.java

```diff
@@ -191,10 +195,8 @@ <T extends Entity> ArrayList<T> processCollectionResponse(Response response)
   private String getEntity(JSONObject jsonObj, HttpUrl url) {
     if(!jsonObj.has(ENTITY)) {
       return getEntityNameFromURL(url);
-    }else if(jsonObj.get("entity").toString().equals("settlement.ondemand")){
-      return "settlement";
-    }else if(jsonObj.get("entity").toString().equals("payment.downtime")){
-      return "payment";
+    }else if(getClass(jsonObj.get("entity").toString()) == null){
```

Comment:
Made this change because it was not able to check those entity properties which has `settlement.ondemand` or `payment.downtime`

response [example](https://razorpay.com/docs/payments/settlements/instant/#using-apis)